lib/shower/index.js

```diff
@@ -1,3 +1,108 @@
-export { default as Container } from './container';
-export { default as defaultOptions } from './default-options';
-export { default as Shower } from './shower';
+import { EventTarget } from '../utils';
+import defaultOptions from './default-options';
+import Slide from './slide';
+import Player from './player';
+import Plugins from './plugins';
+
+const ensureId = (slideElement, index) => {
+    if (!slideElement.id) {
+        slideElement.id = index + 1;
+    }
+
+    return slideElement;
+};
+
+/**
+ * Core module of the Shower.
+ *
+ * @param {object} [options] Shower options.
+ */
+class Shower extends EventTarget {
+    constructor(options) {
+        this.options = Object.freeze({ ...defaultOptions, ...options });
+
+        const containerSelector = this.options.container_selector;
+        this.container = document.querySelector(containerSelector);
+        if (!this.container) {
+            throw new Error(`Shower container with selector '${containerSelector}' not found.`);
+        }
+
+        this._initPlayer();
+        this._initMode();
+        this.plugins = new Plugins(this);
+    }
+
+    _initPlayer() {
+        const slides = [...this.container.querySelectorAll(this.options.slides_selector)]
+            .filter(slideElement => !slideElement.hidden)
+            .map(ensureId)
+            .map(slideElement => new Slide(slideElement, this.options.slide));
+
+        this.player = new Player(slides);
+    }
+
+    _initMode() {
+        const fullMode = this.options.mode_full_classname;
+        const listMode = this.options.mode_list_classname;
+
+        if (this.container.classList.contains(fullMode)) {
+            this.enterSlideMode();
+        } else {
+            this.container.classList.add(listMode);
+        }
+    }
+
+    get isFullMode() {
+        return this._mode === 'full';
+    }
+
+    get isListMode() {
+        return this._mode === 'list';
+    }
+
+    /**
+     * Slide fills the maximum area.
+     */
+    enterFullMode() {
+        if (this.isFullMode) return this;
```

Comment:
`.next()` looks friendlier to me, so if it’s something that might be used in very simple plugins by common uses, it might be an example of good-looking API. I guess I still have jQuery in my ❤️

Reply:
Use cases for fluent method (that always `return this`) are limited to `Player` class.

We wouldn't write something like `shower.exitFullMode().enterFullMode()` or `slide.activate().deactivate().isActive` in a single statement.

The only use case is navigation between slides. I would very much like to avoid making adding `return this` to every method of Shower core because of consistency with one use case in `Player`.

Setter indeed looks odd to people that came to JS before ES5 descriptions.
Let's rename `go` to `goTo` and make `go` work like `history.go`:

```js
player.go(2); // two slides next
player.go(-3); // three slides prev
```

WDYT?

Reply:
Oh, right. I don’t thing we should support chaining, so let’s avoid adding `return this`. I like the idea with `go()` method too.